src/core.jl

```diff
@@ -4,18 +4,26 @@ abstract AbstractPathState
 _insert_and_dedup!(v::Vector{Int}, x::Int) = isempty(splice!(v, searchsorted(v,x), x))
 
 """A type representing a single edge between two vertices of a graph."""
-typealias Edge Pair{Int,Int}
+abstract AbstractEdge
+immutable Edge <: AbstractEdge
```

Comment:
Good that this is immutable, although for 0.6 this will become `immutable struct`